cpp/include/cudf/datetime.hpp

```diff
@@ -207,5 +207,90 @@ std::unique_ptr<cudf::column> is_leap_year(
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
 /** @} */  // end of group
+
+/**
+ * @brief Round up to the nearest day
+ *
+ * @param cudf::column_view of the input datetime values
+ *
+ * @throw cudf::logic_error if input column datatype is not TIMESTAMP
+ * @return cudf::column of the same datetime resolution as the input column
+ */
+std::unique_ptr<cudf::column> ceil_day(
```

Comment:
Do we specifically want a separate API for each resolution? I would assume that a single `ceil(resolution)` function would be easier to use. Even the Python sample in https://github.com/rapidsai/cudf/issues/8682 looks like this.

Reply:
Agree. Then the public API will be similar to the internal detail API that needs an additional parameter indicating which type of time (day/h/m/s etc).

Reply:
I was originally following `_get_dt_field` on the python side, since it broke down the function into separate APIs. I assumed that there was a reason why `_get_dt_field` chose to do that over a single public API that took in an enum?

https://github.com/rapidsai/cudf/blob/branch-21.10/python/cudf/cudf/core/series.py#L6017
https://github.com/rapidsai/cudf/blob/branch-21.10/python/cudf/cudf/_lib/datetime.pyx#L27